.github/workflows/maestro-local.yml

```diff
@@ -74,23 +73,53 @@ jobs:
         with:
           # Ensure we are building the branch and not the branch after being merged on develop
           # https://github.com/actions/checkout/issues/881
-          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.ref }}
+          ref: ${{ github.ref }}
+      - name: Download APK artifact from 'Build APKs' flow
+        uses: actions/download-artifact@v4
+        if: github.event.workflow_run.conclusion == 'success'
+        with:
+          name: elementx-apk-maestro
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
       - name: Download APK artifact from previous job
         uses: actions/download-artifact@v4
+        if: github.event_name == 'workflow_dispatch'
         with:
           name: elementx-apk-maestro
-      - uses: mobile-dev-inc/action-maestro-cloud@v1.9.7
-        if: (github.event_name == 'pull_request' && github.event.pull_request.fork == null) || github.event_name == 'workflow_dispatch'
+      - name: Enable KVM group perms
+        run: |
+          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+          sudo udevadm control --reload-rules
+          sudo udevadm trigger --name-match=kvm
+      - name: Install maestro
+        run: curl -fsSL "https://get.maestro.mobile.dev" | bash
+      - name: Run Maestro tests in emulator
+        uses: reactivecircus/android-emulator-runner@v2
+        env:
+          MAESTRO_USERNAME: maestroelement
+          MAESTRO_PASSWORD: ${{ secrets.MATRIX_MAESTRO_ACCOUNT_PASSWORD }}
+          MAESTRO_RECOVERY_KEY: ${{ secrets.MATRIX_MAESTRO_ACCOUNT_RECOVERY_KEY }}
+          MAESTRO_ROOM_NAME: MyRoom
+          MAESTRO_INVITEE1_MXID: "@maestroelement2:matrix.org"
+          MAESTRO_INVITEE2_MXID: "@maestroelement3:matrix.org"
+          MAESTRO_APP_ID: io.element.android.x.debug
+        with:
+          api-level: ${{ env.API_LEVEL }}
+          arch: ${{ env.ARCH }}
+          profile: ${{ env.DEVICE }}
+          target: ${{ env.TARGET }}
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: true
+          disk-size: 3G
+          script: |
+            chmod +x .github/workflows/scripts/maestro/maestro-local-with-screen-recording.sh
```

Comment:
Is it necessary to set `+x` each time? The file access right can be committed IIRC.